flink-core/src/main/java/org/apache/flink/api/java/typeutils/TypeExtractor.java

```diff
@@ -584,21 +581,6 @@ private static void registerFactory(Type t, Class<? extends TypeInfoFactory> fac
 				// number of parameters the SAM of implemented interface has; the parameter indexing applies to this range
 				final int baseParametersLen = sam.getParameterTypes().length;
 
-				// executable references "this" implicitly
-				if (paramLen <= 0) {
```

Comment:
Why you removed checking the input type information?

Reply:
The input validation caused more errors than it solved. Especially with generic types. For lambdas this validation is limited anyway in a JDK compiler.